### ISSUE TITLE
Cannot install whl-packages with conda-build >3.0.11

### ISSUE THREAD

**Issue by sscherfke** [conda/conda-build#2342]
It looks like conda-build > 3.0.11 can no longer install packages from Python Wheels. 

It looks like conda-build tries to insert these new hashes like crazy everywhere it can.  For example, it renames the srource file `setuptools-36.4.0-py2.py3-none-any.whl` to `setuptools-36_4d54c0bfee.4_4d54c0bfee.0-py2_4d54c0bfee.py3-none-any_4d54c0bfee.whl` which leads to the following error:
```
+ /opt/emsconda/conda-bld/setuptools_1504519488111/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/bin/python -m pip install -I --no-deps setuptools-36_4d54c0bfee.4_4d54c0bfee.0-py2_4d54c0bfee.py3-none-any_4d54c0bfee.whl
setuptools-36_4d54c0bfee.4_4d54c0bfee.0-py2_4d54c0bfee.py3-none-any_4d54c0bfee.whl is not a supported wheel on this platform.
Traceback (most recent call last):
  File "/opt/emsconda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 388, in main
    execute(sys.argv[1:])
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 379, in execute
    noverify=args.no_verify)
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/api.py", line 185, in build
    need_source_download=need_source_download, config=config, variants=variants)
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/build.py", line 1748, in build_tree
    notest=notest,
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/build.py", line 1121, in build
    utils.check_call_env(cmd, env=env, cwd=src_dir)
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/utils.py", line 678, in check_call_env
    return _func_defaulting_env_to_os_environ(subprocess.check_call, *popenargs, **kwargs)
  File "/opt/emsconda/lib/python3.6/site-packages/conda_build/utils.py", line 674, in _func_defaulting_env_to_os_environ
    return func(_args, **kwargs)
  File "/opt/emsconda/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/bin/bash', '-x', '-e', '/opt/emsconda/conda-bld/setuptools_1504519488111/work/conda_build.sh']' returned non-zero exit status 1.
```

**Comment by sscherfke** [comment]
https://github.com/conda/conda-build/pull/2312 this might be the reason.

**Comment by sscherfke** [comment]
I tried adding 
```yaml
conda-build:
    filename_hashing: false
```
to `.condarc`.

This resulted in conda renaming the file to `setuptools-36.4.0-py2.py3-none-any_4d54c0bfee.whl` which still contains a hash and does not work.

**Comment by sscherfke** [comment]
Putting `  fn: setuptools-36.4.0-py2.py3-none-any.whl` into the recipe results in the same behavior (the last hash remains in the filename).

**Comment by msarahan** [comment]
Your issue is a slightly different facet of #2324.  This hash is totally different from the one you've been trying to turn off.  This is the hash of the source file.  You've been trying to turn off the hash of the output package name.  I think the right fix is to append the source hash within the source cache, but then to remove it when placing the source file into the work folder.  I'll do that for the next bug-fix release.

**Comment by sscherfke** [comment]
Yes, the reason for the third hash might be #2324, but the first two hashes in `setuptools-36_4d54c0bfee.4_4d54c0bfee.0-py2_4d54c0bfee.py3-none-any_4d54c0bfee.whl` are replacements of `36` and `4`. Is it by accident that these are the major and minor versions? Or is the 36 handled like the 36 in *py36*?

**Comment by msarahan** [comment]
See https://github.com/conda/conda-build/pull/2330 - this should be fixed in 3.0.14.
